misc-utils/lsfd-sock-xinfo.c

```diff
@@ -404,43 +405,68 @@ struct inet_xinfo {
 	struct in_addr remote_addr;
 };
 
-static bool inet_fill_column(struct proc *proc __attribute__((__unused__)),
-			     struct inet_xinfo *inet,
-			     struct sock *sock __attribute__((__unused__)),
-			     struct libscols_line *ln __attribute__((__unused__)),
-			     int column_id,
-			     size_t column_index __attribute__((__unused__)),
-			     char **str)
+static uint32_t kernel32_to_cpu(enum sysfs_byteorder byteorder, uint32_t v)
 {
-	struct in_addr *nptr = NULL;
-	char s[INET_ADDRSTRLEN];
+	if (byteorder == SYSFS_BYTEORDER_LITTLE)
+		return le32_to_cpu(v);
+	else
+		return be32_to_cpu(v);
+}
 
-	switch(column_id) {
-	case COL_INET_LADDR:
-		nptr = &inet->local_addr;
-		break;
-	case COL_INET_RADDR:
-		nptr = &inet->remote_addr;
-		break;
-	default:
-		return false;
-	}
+/*
+ * L3 abstract-layer stacked on IP and IP6.
+ */
+struct l3_xinfo {
+	struct inet_xinfo inet;
+	unsigned int st;
+};
 
-	if (nptr && inet_ntop(AF_INET, nptr, s, sizeof(s))) {
-		*str = strdup(s);
-		return true;
-	}
-	return false;
-}
+struct l3_xinfo_class {
+	struct sock_xinfo_class sock;
+	struct sock_xinfo *(* scan_line)(const struct l3_xinfo_class *,
+					 char *,
+					 ino_t,
+					 enum sysfs_byteorder);
+	void * (* get_laddr)(struct l3_xinfo *);
+	void * (* get_raddr)(struct l3_xinfo *);
+	bool (* is_any_addr)(void *);
+	int family;
+	size_t size_as_cstring;
```

Comment:
Instead of having to define this member and using `alloca()` I think it would be easier just to use BUFSIZ.

Reply:
Done.